sdk/inc/azure/iot/az_iot_adu_client.h

```diff
@@ -0,0 +1,620 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ *
+ * @brief Definition for the Azure IoT ADU Client
+ *
+ * @note More details about Azure Device Update can be found online
+ * at https://docs.microsoft.com/azure/iot-hub-device-update/understand-device-update
+ *
+ * @note You MUST NOT use any symbols (macros, functions, structures, enums, etc.)
+ * prefixed with an underscore ('_') directly in your application code. These symbols
+ * are part of Azure SDK's internal implementation; we do not document these symbols
+ * and they are subject to change in future versions of the SDK which would break your code.
+ *
+ */
+
+#ifndef _az_IOT_ADU_H
+#define _az_IOT_ADU_H
+
+#include <azure/core/az_json.h>
+#include <azure/core/az_result.h>
+#include <azure/core/az_span.h>
+#include <azure/iot/az_iot_hub_client.h>
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include <azure/core/_az_cfg_prefix.h>
+
+/**
+ * @brief  Define the ADU agent model ID.
+ */
+#define AZ_IOT_ADU_CLIENT_AGENT_MODEL_ID "dtmi:azure:iot:deviceUpdateModel;1"
+
+/**
+ * @brief ADU Agent Version
+ */
+#define AZ_IOT_ADU_CLIENT_AGENT_VERSION "DU;agent/0.8.0-rc1-public-preview"
+
+/**
+ * @brief ADU PnP Component Name
+ */
+#define AZ_IOT_ADU_CLIENT_PROPERTIES_COMPONENT_NAME "deviceUpdate"
+
+/**
+ * @brief ADU Service Response (Accepted)
+ */
+#define AZ_IOT_ADU_CLIENT_REQUEST_ACCEPTED 200
+
+/**
+ * @brief ADU Service Response (Rejected)
+ */
+#define AZ_IOT_ADU_CLIENT_REQUEST_REJECTED 406
+
+/**
+ * @brief ADU Service Action (Apply)
+ */
+#define AZ_IOT_ADU_CLIENT_SERVICE_ACTION_APPLY_DEPLOYMENT 3
+
+/**
+ * @brief ADU Service Action (Cancel)
+ */
+#define AZ_IOT_ADU_CLIENT_SERVICE_ACTION_CANCEL 255
+
+/**
+ * @brief ADU Agent State (Idle)
+ */
+#define AZ_IOT_ADU_CLIENT_AGENT_STATE_IDLE 0
+
+/**
+ * @brief ADU Agent State (In Progress)
+ */
+#define AZ_IOT_ADU_CLIENT_AGENT_STATE_DEPLOYMENT_IN_PROGRESS 6
+
+/**
+ * @brief ADU Agent State (Failed)
+ */
+#define AZ_IOT_ADU_CLIENT_AGENT_STATE_FAILED 255
+
+/**
+ * @brief  Maximum Number of Files Handled by this ADU Agent (Number of URLs)
+ */
+#define AZ_IOT_ADU_CLIENT_MAX_FILE_URL_COUNT 10
+
+/**
+ * @brief  Maximum Number of Files Handled by this ADU Agent (Steps)
+ */
+#define AZ_IOT_ADU_CLIENT_MAX_INSTRUCTIONS_STEPS 10
+
+/**
+ * @brief  Maximum Number of Files Handled by this ADU Agent (File Hashes)
+ */
+#define AZ_IOT_ADU_CLIENT_MAX_FILE_HASH_COUNT 2
```

Comment:
Should this be the same as the URL count?

Reply:
no this is the number of allowed hashes per file (url). In case there is a hash that is not just sha256